R/load_survey.r

```diff
@@ -74,47 +75,47 @@ load_survey <- function(files, ...) {
     }, TRUE)
     merge_files <- names(can_merge[can_merge])
     while (length(merge_files) > 0) {
+
       merged_files <- NULL
       for (file in merge_files)
       {
+        contact_data[[file]] <-
+          contact_data[[file]][, ..merge_id := seq_len(.N)]
         common_id <- intersect(colnames(contact_data[[file]]), colnames(main_surveys[[type]]))
-
-        # is the id unique and can the merge be done uniquely?
-        if (anyDuplicated(contact_data[[file]][, common_id, with = FALSE]) == 0) {
-          test_merge <- merge(
+        merged <- tryCatch({
+          merge(
             main_surveys[[type]], contact_data[[file]], by = common_id,
             all.x = TRUE
           )
-
-          if (nrow(test_merge) > nrow(contact_data[[file]])) {
+        }, error = function(cond) {
+          NULL
```

Comment:
Would it be worth checking which error is returned here?

- if error about cartesian join, ignore and return `NULL`
- if other error, raise it

Reply:
Addressed in 9abb305

I struggled to come up with a test for this. Please propose one if you can think of something.